beet/library/base.py

```diff
@@ -579,9 +585,21 @@ def list_files(
             if extend and not issubclass(content_type, extend):
                 continue
 
-            scope = get_output_scope(
-                content_type.scope, self.pack.pack_format if self.pack else 0
-            )
+            # Use the pack format from the pack's supported_formats.
+            # Otherwise, use the pack_format itself
+            pack_format = 0
+            if self.pack:
+                supported_formats = self.pack.supported_formats
+                if type(supported_formats) is int:
```

Comment:
Using `isinstance(x, int)` instead of comparing the type like `type(x) is int`

also `list[int]` is useless, because :
- at runtime it's just `list`
- at typecheck type, the type narrowing is already `list[int]`

same for FormatsRangeDict, use `dict` + isinstance

Reply:
Also, there is a bunch of places where `get_output_scope` is used with pack_format from a pack nearby.